### PR TITLE
feat(clerk-js,localizations,types): Support for SAML auth flow from s…

### PR DESCRIPTION
**Pull request by yourtallness**
…ign-up & sign-in

## Type of change

- [ ] 🐛 Bug fix
- [X] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [X] `@clerk/types`
- [ ] `@clerk/themes`
- [X] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/components/Shared/SAMLCard.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/Shared/SAMLCard.tsx:1]: 🙃 We can move this under ui/commons

File: packages/types/src/oauth.ts [hunk 2]
```diff
@@ -255,7 +251,28 @@ export type AuthenticateWithRedirectParams = {
...
+      /**
+       * One of the supported OAuth providers you can use to authenticate with, eg 'oauth_google'.
+       */
+      strategy: OAuthStrategy;
+      /**
+       * Email address not expected for OAuth case.
+       */
+      emailAddress: never;
+    }
+  | {
```
Comment (SokratisVidros) [inline packages/types/src/oauth.ts:268]: 🔧 Consider moving this types outside oauth.ts as they are not just about OAuth anymore.

File: packages/clerk-js/src/ui/components/Shared/SAMLCard.tsx (hunk 1 above)
Comment (yourtallness) [reply packages/clerk-js/src/ui/components/Shared/SAMLCard.tsx:1]: Moving to `ui/common`.

File: packages/types/src/oauth.ts (hunk 2 above)
Comment (yourtallness) [reply packages/types/src/oauth.ts:268]: Moving outside of `oauth.ts`.

**Pull request closed without merging**